scalardl/src/scalardl/transfer.clj

```diff
@@ -106,29 +110,31 @@
           (create-asset @client-service id)))))
 
   (invoke! [_ test op]
-    (let [txid (str (java.util.UUID/randomUUID))]
-      (case (:f op)
-        :transfer (let [{:keys [from to amount]} (:value op)
-                        resp (->> (create-argument txid from to amount)
-                                  (.executeContract @client-service
-                                                    "transfer"))]
-                    (if (util/success? resp)
-                      (assoc op :type :ok)
-                      (do
-                        (when (util/unknown? resp)
-                          (warn "The state of transaction is unknown:" txid)
-                          (swap! (:unknown-tx test) conj txid))
-                        (reset! client-service (dl/try-switch-server!
-                                                @client-service test))
-                        (assoc op :type :fail :error (.getMessage resp)))))
-        :check-tx (let [num-committed (check-tx-states test)]
-                    (if num-committed
-                      (assoc op :type :ok :value num-committed)
-                      (assoc op :type :fail :error "Failed to check status")))
-        :get-all (let [balances (read-with-retry @client-service n)]
-                   (if balances
-                     (assoc op :type :ok :value balances)
-                     (assoc op :type :fail :error "Failed to get balances"))))))
+    (case (:f op)
+      :transfer (let [txid (str (java.util.UUID/randomUUID))
+                      {:keys [from to amount]} (:value op)
+                      arg (create-argument txid from to amount)]
+                  (try
+                    (.executeContract @client-service "transfer" arg)
+                    (assoc op :type :ok)
+                    (catch ClientException e
+                      (reset! client-service
+                              (dl/try-switch-server! @client-service test))
+                      (when (util/unknown? e)
```

Comment:
Sorry I'm not really sure but just wondering.
Would it be OK if `ledger` couldn't  return  `UNKNOWN_TRANSACTION_STATUS` for some reason (e.g. failed during returning the status) but the actual transaction status is `UNKNOWN` ?
In such case, does it mark it as failed ?

Reply:
As we discussed, a client should receive `UNKNOWN_TRANSACTION_STATUS ` when gRPC failure like a DL server down or network failure between a client and a server.
That's because a client won't know the contract has succeeded or not.

After fixing Client SDK, it will work well.

Reply:
Thank you for the feedback !
I'm still not sure but should it give `info` when `UNKNOWN_TRANSACTION_STATUS` is returned or it is OK in this case ?

Reply:
In this transfer test, when `UNKNOWN_TRANSACTION_STATUS` is returned, the result is set to be `:fail`.
At the last, `:check-tx` checks the status of transactions that are recorded as `unknown`.
If the status is `committed` in `check-tx`, the count of succeeded transactions is incremented.